### ISSUE TITLE
Pyscripter crashes when it's minimized and RD connection established

### ISSUE THREAD

**Issue by philnagel** [pyscripter/pyscripter#824]
Version: 2.6.0.0
Problem Description: Minimize Pyscripter, lock computer, connect remotely using Microsoft RDP, Pyscripter crashed (message "Pyscripter.exe stopped working")
What did you expect to happen? Expect Pyscripter to stay alive when connecting RD session. Only happens when it's minimized.
What happened instead? Pyscripter crashed instead of staying alive.
Why is this bad/What are the consequences? Possibility to lose unsaved work, nuisance if Pyscripter has to be restarted. Last open tabs/files are lost.
Steps to reproduce the problem: See Problem Description

**Comment by andrewme2** [comment]
I believe I am hitting this too (almost every day).

**Comment by pyscripter** [comment]
Are you using the latest version 3.4?

**Comment by pyscripter** [comment]
I don't used remote desktops but I will have a look.  Is the comment above that the crash occurs only when Pyscripter is minimized correct?

**Comment by andrewme2** [comment]
I have not successfully reproduced the issue on-demand (although I've seen the crash 15-20 times over the past two months), and I was able to just now successfully RDP into my desktop twice without Pyscripter crashing, once with PyScripter in the foreground and once with it minimized.  So, it's still unclear what circumstances lead to the crash.

**Comment by pyscripter** [comment]
Thanks for the information.  Any clues about the circumstances that trigger the error would be helpful in resolving this.   I take it that you don't experience crashes with standard use.

**Comment by andrewme2** [comment]
That's correct. If my memory is correct, I've only seen the crash when logging into my desktop after it has been locked.

**Comment by andrewme2** [comment]
Another clue: the crash usually occurs when I click on the File Explorer tab when I'm currently on the Code Explorer tab (after logging into my laptop in the morning).

**Comment by philnagel** [comment]
Hi! I just remembered about this issue. For some reason, I had not seen it in a while, but it started appearing again. I am using 3.5.1, and whenever I connect to my remote machine and I believe when the screen configuration changes from the last login (for example, I log out from machine with 2 screens, and next log in from machine with 1 screen), Pyscripter gives the message "Pyscripter has stopped working, Windows is looking for a solution to the problem" or similar. 
The only option at that point is to close. So my issue is a little different from andrewme2's, since I can't even do anything again in Pyscripter, it crashed immediately after logging in again.

**Comment by pyscripter** [comment]
>  I log out from machine with 2 screens, and next log in from machine with 1 screen), Pyscripter gives the message "Pyscripter has stopped working, Windows is looking for a solution to the problem" or similar.

Does the error occur when restarting PyScripter after logging out and in again?

**Comment by pyscripter** [comment]
@andrewme2 
Could you please try the executable in this [zip file](https://drive.google.com/file/d/1bTYNHD4WTSjTh8s9TaNs640K7P2glVnQ/view?usp=sharing).  Just replace your PyScripter.exe with the one inside the zip file and report back whether the errors still occur.

**Comment by philnagel** [comment]
@pyscripter I had downloaded this file from #910 (I believe it's the same?) and it seems to fix both issues for me. I would like to do some further testing before I commit to that answer. Is it possible to produce the 32-bit version with those fixes as wells, so I can try in that environment?

**Comment by pyscripter** [comment]
Here are updated links to the 32 bit and 64 bit executables (3.5.2 preview)
- [x64 (64bits)](https://drive.google.com/file/d/1MBIj7YeJAXtAj2yEk5-22GJJraQ5DDVT/view?usp=sharing)
- [x86 (32bits)](https://drive.google.com/file/d/1Bk4OlZ789Xqu6wPjwJEpGCM4Vmp_xtYj/view?usp=sharing)

A lot of features have been added, so please report any other issues that you may find.  This will help make the next release more robust.  Here is the main changes:

History:   v 3.5.2
- New Features
            - Much faster Remote Engine using asynchronous Windows named pipes if pywin32 is available.
            - IDE option to force the use of sockets for connection to the Python server now defaults to False
            - Enhancements to the SSH Engine - now compatible with PuTTY
            - Execute system commands in the interpreter with !. Supports parameter substitution.
            - Clickable status panels with Python version and engine type
            - Text drag & drop between PyScripter and other applications (#554)
            - Triple-click selects line and Quadraple-click selects all
            - Double-click drag selects whole words - triple-click drag selects whole lines
            - Consistent syntax color themes accross supported languages (#855)
- Issues addressed
            - #904, #922, #927, 928, #929, #936

**Comment by andrewme2** [comment]
Sure, I'll try a 3.5.2.  I assume I should use the more recent link you sent (comment immediately above) and not the earlier one from a day ago.

**Comment by pyscripter** [comment]
Yes, please use the most recent ones.

**Comment by andrewme2** [comment]
Update after installing 3.5.2 x64 from the Google Drive link you provided:

I encountered a crash, but I'm not able to determine if it's the same one that I've been encountering each day until now.  Observations:

- on day #2 after installing 3.5.2, PyScripter did *not* crash, even when I navigated the File Explorer within PyScripter

- This morning (Monday), I logged into my laptop and immediately a popup was displayed saying that PyScripter had crashed.  This is different behavior than normal. Typically, PyScripter continues to run OK, and only crashes when I click on the File Explorer

Is there any information that I can grab that I can send you?

**Comment by pyscripter** [comment]
@philnagel Thanks for the feedback.   I also saw your comment at #910.   Did you get any error message in either of the above cases?

**Comment by philnagel** [comment]
> Did you get any error message in either of the above cases?

- In this issue, as soon as I connect to the machine, the generic Windows error message posted by @andrewme2 above (just with Pyscripter instead of Windows Explorer) comes up

- In the other issue #910, the windows essentially freezes, no error shows up. The Pyscripter windows becomes partially opaque or greyed out, and the title bar says "Not Responding". Pyscripter stays like that until you make a mouse click inside of it, then the similar message as above comes up (if I remember right)

> Do you think the two issues are related?

- Not as far as I can tell. I can reproduce this issue easily with the steps I described in an earlier message (connect remotely with different monitor setup, while Pyscripter is in the background)

- The other issue seems very random. I think it may be somehow related to code complettion as you type - can this be changed to complete only on tab like some other IDEs have it? Although the as-you-type feature is one of the ones that makes Pyscripter so much better, it could be worth to test this theory.

> Did you set the DE option, Tools, Options, IDE Options, File Explorer, Background processing. If not please set it to False and report whether it makes a difference.

- Not yet. I will set this and continue testing and report back.


> Did you have any other issues/regressions with this release?

- The only issue I noticed is that it seems part of my settings in the ini file are ignored, specifically my custom tools and code templates do not show up anymore.

**Comment by pyscripter** [comment]
- You turn "Complete as you type" off through Tools, Options IDE Options, Code Completion.  You may want to have a look at this [article](https://pyscripter.blogspot.com/2011/08/code-completion-improvements.html) which explains some of the other options.
- More likely to be a problem are Code hints since they are constantly evaluated as you move the mouse.  You can disable those through Tools, Options, IDE Options, Editor.
- Regarding the ignored custom tools and code templates could you please post here or email pyscripter at gmail your pyscripter.ini file so that I can have a look at this problem?
-Finally regarding this issue the steps to reproduce are the same as those described at the top of this page.  Is this right?  You do not get this issue if PyScripter is not minimized.

**Comment by pyscripter** [comment]
Just tried to reproduce this issue.  Have not used RD before.  So my Desktop has a standard DPI (scale 100%) and I am connected to it from a High DPI laptop.  PyScripter was minimized on the Desktop when I connected.  No crash.  Unfortunate I cannot try the reverse since my laptop has the Home Edition of Windows which does not support RD.

By the way RD looks good!

**Comment by philnagel** [comment]
I will do some different scenarios tomorrow and see if I can get the exact one when this happens.
Yes, RD is very convenient when you have more than one machine, makes things very easy!

**Comment by andrewme2** [comment]
Good news, I have not had any issues since I turned on Background Processing.  I can navigate using the File Explorer widget and no crashes occur, even after several days of running.

By the way, the only oddity I've encountered with this 3.5.2 pre-release is that I can't drag/drop files from Windows File Explorer (outside of Pyscripter) into PyScripter.  I believe that is a regression, since I'm pretty sure I use that feature multiple times per week.

**Comment by pyscripter** [comment]
> Good news, I have not had any issues since I turned on Background Processing. I can navigate using the File Explorer widget and no crashes occur, even after several days of running.

@andrewme2 Glad to hear that.  I suppose you mean **turned off Background Processing**.   I will change the default to False to be on the safe side.  Does your comment also apply for this issue (Remote Desktop use)??

Regarding the Drag/drop of files this may be an interaction with this:
- Text drag & drop between PyScripter and other applications (#554)
I will have a look.

**Comment by andrewme2** [comment]
Yes, I meant turned "off" background processing. Thanks for clarifying.  And yep, I RDP'd into my laptop twice (on separate days) and I did not encounter a crash either time.

**Comment by pyscripter** [comment]
@andrewme2 Until this get fixed you can still drop files on the tab bar but not inside the editor.

**Comment by philnagel** [comment]
Here are the specific steps I can take to always make this issue appear:

1. Connect using Microsoft Remote Desktop to host machine.

2. Open PyScripter on host machine, make sure the PyScripter window is maximized, not only using the partial screen

2b. Now you can either leave PyScripter running in the foreground, leave it running in the background, or minimize it.

3. Disconnect RD session while leaving PyScripter running.

4. On client machine (machine you are connecting to host from) change the display resolution. Change to any other value available for your display.

5. Connect to host machine via Microsoft Remote Desktop client. PyScripter will crash.

Would you like me to revise my first post in this issue to reflect this?

**Comment by pyscripter** [comment]
@philnagel  I tried the exact steps but I could reproduce the crash.  I have made nevertheless a small change that might have an impact when PyScripter is minimised.  I will make a new release and you can try with the new release again.

**Comment by philnagel** [comment]
Sounds good. Thank you for all your work trying to find this bug. 
Is the new beta 3.6 the one you're referring to?

**Comment by pyscripter** [comment]
Yes.

**Comment by pyscripter** [comment]
Steps taken to prevent Remote Desktop connection/disconnection issues.  Please check the forthcoming version.
